src/main/java/com/google/api/generator/gapic/composer/common/AbstractTransportServiceStubClassComposer.java

```diff
@@ -473,8 +477,7 @@ protected abstract Expr createTransportSettingsInitExpr(
         argList ->
             NewObjectExpr.builder().setType(creatorMethodReturnType).setArguments(argList).build();
 
-    TypeNode stubSettingsType =
-        typeStore.get(getTransportContext().classNames().getServiceStubSettingsClassName(service));
+    TypeNode stubSettingsType = typeStore.get(ClassNames.getServiceStubSettingsClassName(service));
```

Comment:
Why is this getting replaced? Is it solely because a static method is accessed via instance reference? 

Note, since we already have `ClassNames` as part of transport context, all the static usages of ClassNames are potentially error-prone (a change in context one will not affect places, which are potentially the subject to change, because they are called statically), so basiclaly we need to convert it the other way around to reach consistency -> convert all Static ClassNames calls to dynamic instance-level ones (and making the method non-static).

The reason why those static methods are still static is historical.